internal/provider/resource_instance.go

```diff
@@ -669,18 +687,26 @@ func (r *instanceResource) Update(ctx context.Context, req resource.UpdateReques
 	// Due to the design of the API, when updating an instance all
 	// parameters must be set. This means we set all of the parameters
 	// even if only a single one changed.
-	if state.BootDiskID != plan.BootDiskID ||
+	if state.AutoRestartPolicy != plan.AutoRestartPolicy ||
+		state.BootDiskID != plan.BootDiskID ||
 		state.Memory != plan.Memory ||
 		state.NCPUs != plan.NCPUs {
 
 		params := oxide.InstanceUpdateParams{
 			Instance: oxide.NameOrId(state.ID.ValueString()),
 			Body: &oxide.InstanceUpdate{
-				BootDisk: oxide.NameOrId(plan.BootDiskID.ValueString()),
-				Memory:   oxide.ByteCount(plan.Memory.ValueInt64()),
-				Ncpus:    oxide.InstanceCpuCount(plan.NCPUs.ValueInt64()),
+				Memory: oxide.ByteCount(plan.Memory.ValueInt64()),
+				Ncpus:  oxide.InstanceCpuCount(plan.NCPUs.ValueInt64()),
 			},
 		}
+		if !plan.AutoRestartPolicy.IsNull() {
+			autoRestartPolicy := oxide.InstanceAutoRestartPolicy(plan.AutoRestartPolicy.ValueString())
+			params.Body.AutoRestartPolicy = &autoRestartPolicy
```

Comment:
Totally optional but if you wanted to avoid the type shuffling and variable creation. I know the type shuffling is a bit annoying.

```suggestion
			params.Body.AutoRestartPolicy = (*oxide.InstanceAutoRestartPolicy)(plan.AutoRestartPolicy.ValueStringPointer())
```